docs/experiment_config_files.rst

```diff
@@ -85,7 +85,7 @@ This specifies the model architecture. An typical example looks like this
     decoder: !MlpSoftmaxDecoder
```

Comment:
For a while I've disliked the name `MlpSoftmaxDecoder`. This seems to be a good time to change it, because we're changing lots of things. This would also probably allow us to kill this issue:
> https://github.com/neulab/xnmt/issues/382

I'd suggest maybe for the decoder having `mlp:` and `softmax:` specified separately, where the MLP layer is just in charge of transforming one vector to another, and the softmax calculates the probabilities, n-best, etc.

Also, I think it'd be a lot better if we could separate out the attention from this. These are all independent things, and having them in one big package seems to be limiting. Or alternatively, we could have one big package (`AttentionalOutput`?) that has each of these elements as a component.